txt2html.py

```diff
@@ -3,6 +3,46 @@
 import os
 import shutil
 import argparse
+import re
+
+def contains_italics(word):
+    # Markdown Pattern Regular Expressions
+
+    # Matches *word*, *WORD*, *woRd*, **word**
+    italic_pattern1 = r'(?<!\*)\*(?:\*|[^*]+)\*(?!\*)'
+
+    # Matches _word_, _WORD_, _woRd_, __word__
+    italic_pattern2 = r'(?<!\_)_(?:\_|[^*]+)_(?!\_)'
+
+    # Return True if word matches either RegEx pattern, False otherwise
+    return (re.search(italic_pattern1, word) or re.search(italic_pattern2, word))
+
+def process_line(file_line):
+
+
+    # Split updatedLine into words
+    words = file_line.split()
+
+    # Temporary line
+    modifiedLine = ""
+    for word in words:
+        # This if/else structure checks if the word matches a Markdown regex pattern (italics only for now)
+        # If the word matches a Markdown regex it is modified with appropriate HTML tags
+        
+        # Check if word matches either italic regex pattern
+        if contains_italics(word):
+            # Replace beginning and ending '*' or "_" with <i>...</i> tags
+            # Examples: 
+            #   *word* -> <i>word</i>
+            #   _word_ -> <i>word</i>
+            #   _word* -> _word*
+            #   __word__ -> <i>_word_</i>
```

Comment:
can we omit `__word__` situation based on `contains_italics` function comments?

Reply:
Currently, **bold** conversion support (converting `**word**` or `__word__` to `<b>world</b>`) has not been implemented in the program.
Because of this, although situations like `__word__`  or `**word**` _should_ return `<b>word</b>`, they will currently return `<i>_word_</i>` and `<i>*word*</i>` instead

This behavior is inherent in the `italic_pattern1` and `italic_pattern2` regular expressions in `contains_italics`, which allows for asterisks ( `*`s) and underscores ( `_`s) to exist between the wrapping asterisks and underscores.
This allows conversions such as:
1. `*w_rd* to <i>w_rd</i>`
2.  `_w_rd_ to <i>w_rd</i>`, and also
3.  `__word__` to `<i>_word_</i>` and `**word**` to `<i>*word*</i>` among other things.

The conversions shown in **3** are currently unintended, but will be eliminated once **bold** conversion support is added.

Reply:
The conversions shown in **3** still happen. However, they can be prevented by adding bold Markdown conversion support.

I've added TO-DO comments in txt2html.py that detail **one way** as to implementing bold Markdown conversions and preventing `__word__` to `<i>_word_</i>` and `**word**` to `<i>*word*</i>` conversions.
The process is as follows:
- Implement a method called `contains_bold(word)` that checks in `word` matches regex patterns for bold Markdown syntax. The regex patterns and logic in `contains_italics(word)` can be used as a reference
- In `process_line`, implement the conversion of bold Markdown syntax to bold HTML (line 44)
- In `process_line`, first check if the word contains bold Markdown syntax before checking for italics Markdown syntax.